src/scope/mod.rs

```diff
@@ -231,13 +263,16 @@ impl<'scope> Scope<'scope> {
         unsafe {
             let old_value = self.counter.fetch_add(1, Ordering::SeqCst);
             assert!(old_value > 0); // scope can't have completed yet
+
             let job_ref = Box::new(HeapJob::new(self, body)).as_job_ref();
             let worker_thread = WorkerThread::current();
             if !worker_thread.is_null() {
                 let worker_thread = &*worker_thread;
                 let spawn_count = worker_thread.spawn_count();
                 spawn_count.set(spawn_count.get() + 1);
                 worker_thread.push(job_ref);
+            } else if let Some(thread_pool) = self.thread_pool {
```

Comment:
So I'm a bit confused here. I think you could do `thread_pool.install(|| scope(|s| ...))` and you would be all set, no? 

The idea is that, when you call `install`, you are in the worker thread for this thread-pool, so then when we come into `scope`, we are already there.